presto-main/src/main/java/com/facebook/presto/operator/window/WindowPartition.java

```diff
@@ -39,7 +43,7 @@
     private final int partitionEnd;
 
     private final int[] outputChannels;
-    private final List<WindowFunction> windowFunctions;
+    private final Map<WindowFunction, FrameInfo> windowFunctions;
```

Comment:
Same, it seems like we should not be using a Map here (especially since that forces WindowFunction implementations to have a valid equals and hashCode implementation). I would prefer either parallel lists or a wrappered object over the 2 things here.